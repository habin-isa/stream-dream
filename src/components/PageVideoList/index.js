import React, { useState, useCallback, useEffect } from "react";

import VideoItem from "../VideoItem";
import { getVideos } from "../../services/videos";

const PageVideoList = () => {
  const [videoData, setVideoData] = useState([]);

  const loadVideos = useCallback(async () => {
    try {
      const response = await getVideos({
        params: {
          type: "video",
          part: "snippet",
          q: "PAQ"
        }
      });
      setVideoData(response.data.items);
      console.log(videoData);
    } catch (error) {
      console.log("error fetching videos");
    } finally {
      console.log("loadVideos has ran");
    }
  }, [videoData]);

  useEffect(() => {
    loadVideos();
  }, [loadVideos]);

  const renderedVideos = videoData.map(video => {
    return <VideoItem key={video.id.videoId} video={video} />;
  });

  return (
    <div>
      FERRARI
      {renderedVideos}
    </div>
  );
};

export default PageVideoList;
