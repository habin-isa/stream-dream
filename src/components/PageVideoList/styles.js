import styled from 'styled-components';

export const Wrapper = styled.div`
    background: black;
`;

export const Menu = styled.div`
    display: flex;
    width: 100%;
    font-weight: bold;
    cursor: pointer;
    text-align: center;
`;

export const DateBox = styled.div`
    background: white;
    width: 100%;
    padding: 10px;
    &:hover {
        background: #fcba03;
    }
`;

export const CountBox = styled.div`
    background: black;
    color: white;
    width: 100%;
    padding: 10px;
    &:hover {
        background: #fcba03;
    }
`;
