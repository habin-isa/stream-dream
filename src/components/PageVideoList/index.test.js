import React from 'react';
import { render } from '@testing-library/react';
import PageVideoList from '.';

describe('PageVideoList', () => {
    it('should render', () => {
        const { container } = render(<PageVideoList />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
