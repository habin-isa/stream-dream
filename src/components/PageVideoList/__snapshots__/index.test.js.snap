// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageVideoList should render 1`] = `
<div
  class="sc-bdVaJa cDyoBV"
/>
`;
